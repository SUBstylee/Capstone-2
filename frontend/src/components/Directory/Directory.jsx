import React from "react";
import MenuItem from "../MenuItem/MenuItem";
import './Directory.scss';

const Directory =()=> {
    const cats = {
            sections: [
                {
                    title: 'hats',
                    imageUrl: '/assets/images/shop-img/hats.png',
                    id: 1,
                    linkUrl: '__Hats__'
                },
                {
                    title: 'jackets',
                    imageUrl: '/assets/images/shop-img/jackets.png',
                    id: 2,
                    linkUrl: '__Jackets__'
                },
                {
                    title: 'shoes',
                    imageUrl: '/assets/images/shop-img/sneakers.png',
                    id: 3,
                    linkUrl: '__Shoes__'
                },
                {
                    title: 'women',
                    imageUrl: '/assets/images/shop-img/womens.png',
                    id: 4,
                    linkUrl: '__Women__'
                },
                {
                    title: 'men',
                    imageUrl: '/assets/images/shop-img/men.png',
                    id: 5,
                    linkUrl: '__Men__'
                },
                {
                    title: 'all',
                    imageUrl: '/assets/images/shop-img/sample.png',
                    id: 6,
                    linkUrl: '__All__'
                }
            ]
        };

    return (
        <div className="Directory">
            {
                cats.sections.map(({ id, ...others }) => (
                    <MenuItem key={id} {...others} />
                ))
            }
        </div>
    );
};

export default Directory;