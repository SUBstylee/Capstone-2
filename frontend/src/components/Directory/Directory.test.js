import { render } from '@testing-library/react';
import Directory from './Directory.jsx';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

test("renders Directory component without crashing", () => {
    const { getByText } = render(
        <Provider store={store}><BrowserRouter><Directory /></BrowserRouter></Provider>
    );

    expect(getByText('HATS' && 'SHOES' && 'JACKETS' && 'MEN' && 'WOMEN' && 'ALL')).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}><BrowserRouter><Directory /></BrowserRouter></Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});