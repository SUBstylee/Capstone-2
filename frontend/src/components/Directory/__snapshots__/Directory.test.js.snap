// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Directory"
  >
    <div
      class="MenuItem"
    >
      <div
        class="background-image"
        style="background-image: url(/assets/images/shop-img/hats.png);"
      />
      <div
        class="content"
      >
        <h1
          class="title"
        >
          HATS
        </h1>
        <span
          class="subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      class="MenuItem"
    >
      <div
        class="background-image"
        style="background-image: url(/assets/images/shop-img/jackets.png);"
      />
      <div
        class="content"
      >
        <h1
          class="title"
        >
          JACKETS
        </h1>
        <span
          class="subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      class="MenuItem"
    >
      <div
        class="background-image"
        style="background-image: url(/assets/images/shop-img/sneakers.png);"
      />
      <div
        class="content"
      >
        <h1
          class="title"
        >
          SHOES
        </h1>
        <span
          class="subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      class="MenuItem"
    >
      <div
        class="background-image"
        style="background-image: url(/assets/images/shop-img/womens.png);"
      />
      <div
        class="content"
      >
        <h1
          class="title"
        >
          WOMEN
        </h1>
        <span
          class="subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      class="MenuItem"
    >
      <div
        class="background-image"
        style="background-image: url(/assets/images/shop-img/men.png);"
      />
      <div
        class="content"
      >
        <h1
          class="title"
        >
          MEN
        </h1>
        <span
          class="subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
    <div
      class="MenuItem"
    >
      <div
        class="background-image"
        style="background-image: url(/assets/images/shop-img/sample.png);"
      />
      <div
        class="content"
      >
        <h1
          class="title"
        >
          ALL
        </h1>
        <span
          class="subtitle"
        >
          SHOP NOW
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
